### PR TITLE
Add VSIX extension

### PR DESCRIPTION
**Pull request by lanfeust69**
Visual Studio 2015 deprecates AddIns in favor of VSIX extensions, so we cannot
integrate GitExtensions in it.

The new project replicates most of the AddIn features.

TODO :
 + handle the "Commit (branch_name)" caption of the commit command.
 + see if installing the extension should be part of the setup.

### CODE REVIEW

File: GitExtensions.VS2015.sln [hunk 1]
```diff
@@ -0,0 +1,563 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (sharwell) [inline GitExtensions.VS2015.sln]: :question: Why put this in a separate solution file? Only two solution files are necessary: one for Windows, and one for Mono.

**Comment by sharwell** [comment]
The biggest question here is why use an add-in for three versions and a VSIX which duplicates the functionality in a fourth? We can use one VSIX which targets Visual Studio 2010 through 2015, replace the add-in, and be done with it.

**Comment by sharwell** [comment]
@lanfeust69 One option is creating the **.csproj** file for the new extension, but intentionally not including the new **.sln** file. I could come along after and modify the project to work with multiple versions of Visual Studio instead of just 2015. The key will be making sure all the features are still supported.

**Comment by lanfeust69** [comment]
@sharwell : I agree completely that removing the AddIn, and having only the VSIX (working on all supported versions of Visual Studio) should be the target.

Unfortunately, I don't really have the time (at least not before one or two weeks) to :
- see what would be needed for backward compatibility of the VSIX
- figure out the complete set of features of the AddIn
- find how to implement the known missing feature (the name of the current branch next to the commit button), which doesn't seem completely straightforward

I wanted to quickly have an integration in VS2015 (because I was already really missing it). I think what I did is a starting point, and the sooner it is public, the smaller the chances that someone else wastes time doing it again.

If I understood correctly, you'd rather not have the VS2015 **.sln** in the commit, so that there is no temptation to fragment the code depending on the platform, Is that so ? If yes, just confirm, and I'll amend the pull request to remove it (and add an explanation in the commit message)...

**Comment by sharwell** [comment]
> I wanted to quickly have an integration in VS2015 (because I was already really missing it). I think what I did is a starting point, and the sooner it is public, the smaller the chances that someone else wastes time doing it again.

Works for me. I'd rather start with your work when it comes to implementing it for all platforms.

**Comment by sharwell** [comment]
@lanfeust69 What about if tonight I send a pull request to your fork with the changes to make the project work in multiple versions of VS? If you accept/merge that pull request, *this* pull request would be automatically updated to include those changes so when it's merged into Git Extensions it's ready to work.

My goal:

* VSIX loads and builds in all supported versions of VS, allowing us to later make rapid progress towards feature parity
* Not necessarily all features are implemented, so the installer would only actually deploy it to 2015 for now

**Comment by lanfeust69** [comment]
@sharwell : Great !

I may be offline for a while, but I'll surely be able to connect before tomorrow evening.

**Comment by lanfeust69** [comment]
@KindDragon : It seems TeamCity build chokes on this. There are two issues :
- GitExtensions still uses the "old" [NuGet restore](https://docs.nuget.org/consume/package-restore). Of course I could make GitExtensionsVSIX work the same way, but I think it would be better to move along, which means adding a [restore step](http://blog.jetbrains.com/teamcity/2013/08/nuget-package-restore-with-teamcity/).
- The ProjectReference to GitPlugin doesn't seem to be taken into account, I can't see why. The csc.exe command doesn't reference it, but it works fine (of course !) on my machine. Is there a way to exactly reproduce TeamCity build environment ?

**Comment by vbjay** [comment]
They had a restore step and it was causing issues.

**Comment by linquize** [comment]
@KindDragon Could you install Visual Studio SDK on TeamCity?

**Comment by KindDragon** [comment]
No, they doesn't have VS installed on CodeBetter TC

**Comment by KindDragon** [comment]
Because of that I've used separate solution configuration for TC

**Comment by linquize** [comment]
Only install .net framework 4 with built-in C# compiler in teamcity?

**Comment by KindDragon** [comment]
> Only install .net framework 4 with built-in C# compiler in teamcity?

I don't know exactly http://codebetter.com/codebetter-ci/

Maybe some of required assemblies available in NuGet

**Comment by linquize** [comment]
It is very likely the server only has .net framework 4 (MSBuild 4, C# 4 compiler) or 4.5 (MSBuild 4, C# 5 compiler).
It may not install MSBuild 12 or MSBuild 14 (C# 6 compiler)

**Pull request merged**